generators/app/templates/src/StarterKit/DataAccess/Repositories/EntityRepositoryBase.cs

```diff
@@ -0,0 +1,273 @@
+      Func<IQueryable<TEntity>, IQueryable<TEntity>> includes = null)
+    {
+      var result = QueryDb(filter, includes, sortString);
+      return await result.ToListAsync();
+    }
+
+    public virtual void Load(Expression<Func<TEntity, bool>> filter,
+      string sortString = null,
+      Func<IQueryable<TEntity>, IQueryable<TEntity>> includes = null)
+    {
+      var result = QueryDb(filter, includes, sortString);
+      result.Load();
+    }
+
+    public virtual async Task LoadAsync(Expression<Func<TEntity, bool>> filter,
+      string sortString = null,
+      Func<IQueryable<TEntity>, IQueryable<TEntity>> includes = null)
+    {
+      var result = QueryDb(filter, includes, sortString);
+      await result.LoadAsync();
+    }
+
+    public virtual IEnumerable<TEntity> QueryPage(int startRow, int pageLength, Expression<Func<TEntity, bool>> filter,
+      string sortString = null,
+      Func<IQueryable<TEntity>, IQueryable<TEntity>> includes = null)
+    {
+      var result = QueryDb(filter, includes, sortString);
+      return result.Skip(startRow).Take(pageLength).ToList();
+    }
+
+    public virtual async Task<IEnumerable<TEntity>> QueryPageAsync(int startRow, int pageLength,
+      Expression<Func<TEntity, bool>> filter,
+      string sortString = null,
+      Func<IQueryable<TEntity>, IQueryable<TEntity>> includes = null)
+    {
+      var result = QueryDb(filter, includes, sortString);
+      return await result.Skip(startRow).Take(pageLength).ToListAsync();
+    }
+
+    public virtual void Add(TEntity entity)
+    {
+      if (entity == null) throw new InvalidOperationException("Unable to add a null entity to the repository.");
+      Context.Set<TEntity>().Add(entity);
+    }
+
+    public virtual void AddBatch(IEnumerable<TEntity> entities)
+    {
+      if (entities == null) throw new ArgumentNullException(nameof(entities));
+      Context.Set<TEntity>().AddRange(entities);
+    }
+
+    public virtual TEntity Update(TEntity entity)
+    {
+      return Context.Set<TEntity>().Update(entity).Entity;
+    }
+
+    public virtual IEnumerable<TEntity> UpdateBatch(IEnumerable<TEntity> entities)
+    {
+      if (entities == null) throw new ArgumentNullException(nameof(entities));
+      return entities.Select(Update);
+    }
+
+    public virtual void Remove(TEntity entity)
+    {
+      Context.Set<TEntity>().Attach(entity);
+      Context.Entry(entity).State = EntityState.Deleted;
+      Context.Set<TEntity>().Remove(entity);
+    }
+
+    public virtual void RemoveBatch(IEnumerable<TEntity> entities)
+    {
+      if (entities == null) throw new ArgumentNullException(nameof(entities));
+      Context.Set<TEntity>().RemoveRange(entities);
+    }
+
+    public virtual void Remove(TId id)
+    {
+      Remove(Get(id));
+    }
+
+    public virtual void RemoveBatch(IEnumerable<TId> ids)
+    {
+      if (ids == null) throw new ArgumentNullException(nameof(ids));
+      foreach (var id in ids)
+      {
+        Remove(id);
```

Comment:
kan je niet beter een select op basis van id doen en daarna removerange aanroepen? Deze manier van werken gaat resulteren in aparte get en delete queries voor elke id, wat resulteert in 2 db operaties per foreach iteratie, terwijl je dit zonder foreach met 2 queries zou kunnen doen voor de ganse lijst in 1 keer

Reply:
Terechte opmerking, is aangepast.